### ISSUE TITLE
Incorrect error: expected `Pin<Box<dyn Future>>>`, found `Pin<Box<impl Future>>>`

### ISSUE THREAD

**Issue by arilotter** [rust-lang/rust-analyzer#12145]
**rust-analyzer version**: 2022-04-11

**rustc version**: rustc 1.61.0-nightly (76d770ac2 2022-04-02)

**relevant settings**: 

**repro**:
```rs
use core::{future::Future, pin::Pin};

fn test() -> Pin<Box<dyn Future<Output = ()>>> {
    Box::pin(async move {})
}

fn main() {}

```

Using `cargo check`, works fine.
In RA, I see `expected Pin<Box<dyn Future<Output = ()>, Global>>, found Pin<Box<impl Future<Output = ()>, Global>> rust-analyzer(type-mismatch)` underneath the `Box::pin(async move {})`.

**Comment by flodiebold** [comment]
Duplicate of #11815.
